packages/compass-saved-aggregations-queries/src/hooks/use-grid-filters.tsx

```diff
@@ -0,0 +1,278 @@
+      )}
+      onChange={onSelect}
+      defaultValue={defaultValue}
+    >
+      {items.map((item) => (
+        <Option key={item.value} value={item.value}>
+          {item.value}
+        </Option>
+      ))}
+    </Select>
+  );
+};
+
+function useSearchFilter(): [React.ReactElement, string] {
+  const ref = useRef<HTMLInputElement>(null);
+  const [search, setSearch] = useState('');
+  useEffect(() => {
+    ref.current?.focus();
+  }, [search]);
+  const searchControls = useMemo(() => {
+    return (
+      <TextInput
+        ref={ref}
+        className={searchInputStyles}
+        aria-label="Search"
+        type="search"
+        placeholder="Search"
+        value={search}
+        defaultValue={search}
+        onChange={(e) => {
+          setSearch(e.target.value);
+        }}
+        spellCheck={false}
+      />
+    );
+  }, [search]);
+
+  return [searchControls, search];
+}
+
+function useSelectFilter(tree: Tree[]): [React.ReactElement, SelectState] {
+  const [selectState, setSelectState] = useState<SelectState>({});
+  const [collections, setCollections] = useState<Tree[]>([]);
+
+  const selectDatabase = useMemo(() => {
+    return (database: string): void => {
+      setSelectState({
+        database: database ?? undefined,
+      });
+      const items = tree.find((x) => x.value === database)?.items;
+      setCollections(items || []);
+    };
+  }, [tree]);
+
+  const selectCollection = useMemo(() => {
+    return (collection: string): void => {
+      setSelectState({
+        ...selectState,
+        collection: collection ?? undefined,
+      });
+    };
+  }, [selectState]);
+
+  const treeControls = useMemo(() => {
+    return (
+      <div className={selectContainer}>
+        <RenderFilterSelect
+          items={tree}
+          placeHolder={'All databases'}
+          onSelect={selectDatabase}
+          defaultValue={selectState.database}
+        />
+        <RenderFilterSelect
+          items={collections}
+          placeHolder={'All collections'}
+          onSelect={selectCollection}
+          defaultValue={selectState.collection}
+        />
+      </div>
+    );
+  }, [selectDatabase, tree, collections, selectCollection, selectState]);
+
+  return [treeControls, selectState];
+}
+
+function convertItemsToTree(items: Item[]): Tree[] {
+  const dbCollectionMap: Record<string, string[]> = {};
+  const tree = [];
+  items.forEach((item) => {
+    if (!dbCollectionMap[item.database]) {
+      dbCollectionMap[item.database] = [];
+    }
+    dbCollectionMap[item.database].push(item.collection);
+  });
+  for (const database in dbCollectionMap) {
+    const collections = dbCollectionMap[database]
+      .filter((collection, index, arr) => arr.indexOf(collection) === index)
+      .map((collection) => ({
+        value: collection,
+      }));
+
+    tree.push({
+      value: database,
+      items: collections,
+    });
+  }
+  return tree;
+}
```

Comment:
I don't mind that Tree type and conversion logic, but it also feels a bit unneccesary unless I'm missing something. Nothing except the select filter needs this structure and even there it can be derived in place without the conversion as something like

```
const dbs = unique(items.map(item => item.database))
const collections = unique(items.filter(item => selectedDatabase ? item.database === selectedDatabase : true).map(item => item.collection))
```